src/bin/pg_autoctl/pgsetup.c

```diff
@@ -867,7 +860,7 @@ pg_setup_get_username(PostgresSetup *pgSetup)
 {
 	uid_t uid;
 	struct passwd *pw;
-	char *userEnv;
+	char userEnv[NAMEDATALEN];
```

Comment:
Now we have a double copy of the userEnv name, local buffer and then strdup() to the caller. It it worth changing the API to make a single copy somehow?

Reply:
I think it's fine for now. I never like changing an API for only a single usecase, because you don't know what a good abstraction level is at that point. If this happens more often in the future we can add a helper then.